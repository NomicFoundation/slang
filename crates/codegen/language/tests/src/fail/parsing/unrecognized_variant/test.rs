#![allow(unused_crate_dependencies)]

codegen_language_macros::compile!(Language(
    name = Foo,
    root_item = Bar1,
    leading_trivia = Sequence([]),
    trailing_trivia = Sequence([]),
    versions = ["1.0.0"],
    sections = [Section(
        title = "Section One",
        topics = [
            Topic(
                title = "Topic One",
                items = [
                    Struct(name = Bar1, fields = (field = Required(Bar2))),
                    Struct(name = Bar2, fields = (field = Required(Bar1)))
                ]
            ),
            Topic(title = "Topic Two", items = [Unrecognized(true)])
        ]
    )]
));

fn main() {}
