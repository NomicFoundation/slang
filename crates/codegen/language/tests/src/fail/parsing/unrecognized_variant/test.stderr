error: Expected a variant: 'Struct' or 'Enum' or 'Repeated' or 'Separated' or 'Precedence' or 'Trivia' or 'Keyword' or 'Token' or 'Fragment'
  --> src/fail/parsing/unrecognized_variant/test.rs:19:49
   |
19 |             Topic(title = "Topic Two", items = [Unrecognized(true)])
   |                                                 ^^^^^^^^^^^^
